polymorphia-backend/src/main/java/com/agh/polymorphia_backend/dto/item/ItemDto.java

```diff
@@ -0,0 +1,40 @@
+package com.agh.polymorphia_backend.dto.item;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+
+@SuperBuilder
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@AllArgsConstructor
+@NoArgsConstructor
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.EXISTING_PROPERTY,
+        property = "type",
+        visible = true
+)
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = FlatBonusItemDto.class, name = "FLAT_BONUS"),
+        @JsonSubTypes.Type(value = PercentageBonusItemDto.class, name = "PERCENTAGE_BONUS")
+})
+public class ItemDto {
```

Comment:
```suggestion
public class RequestItemDto {
```

I guess it's not necessary right now but we'll probably use different DTOs for requests and responses with different data inside so we might start use different names for them now.

We might need more DTOs for the same type in the future (e.g. for different endpoints) but we probably shouldn't get ahead of ourselves.